### PR TITLE
Hotfix/make erd docker login

### PR DESCRIPTION
**Pull request by andrew-jameson**
## Summary of Changes
Adds `docker-login` ahead of docker pull usage per #3165 and [this thread](https://mattermost.teamraft.com/goraft/pl/oarxqqpeqjb1px84kjp91bpbjw) from @ADPennington. Also addressed an unseen technical debt issue that pipeline did not fail when artifact was not found.
Screenshot 2024-10-11 at 1 44 35 PM

## How to Test
[Latest pipeline](https://app.circleci.com/pipelines/github/raft-tech/TANF-app/29842/workflows/cfe7565e-3435-4a73-aace-de2f5c28a896/jobs/87712) shows passing `make_erd` and artifact now [exists](https://output.circle-artifacts.com/output/job/9a766af7-da16-4204-9d12-4f249135bcfe/artifacts/0/tdrs-backend/tdp_erd.png). 

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] [**_insert ACs here_**]
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: .circleci/build-and-test/workflows.yml [hunk 1]
```diff
@@ -49,6 +49,13 @@
                 - /^release.*/
           requires:
             - secrets-check
+      - make_erd: # from ../util folder
+          filters:
+            branches:
+              only:
+                - develop
+                - master
+                - hotfix/make-erd-docker-login
```
Comment (andrew-jameson) [inline .circleci/build-and-test/workflows.yml]:
```suggestion
```
To be removed before merge.

File: .circleci/deployment/workflows.yml [hunk 2]
```diff
@@ -148,3 +148,4 @@
               only:
                 - develop
                 - master
+                - hotfix/make-erd-docker-login
```
Comment (andrew-jameson) [inline .circleci/deployment/workflows.yml]:
```suggestion
```
To be removed before merge.

File: tdrs-backend/setup.cfg [hunk 3]
```diff
@@ -6,6 +6,7 @@ omit =
     tdpservice/settings/production.py
     tdpservice/settings/staging.py
     tdpservice/wsgi.py
+    tdpservice/scheduling/*
```
Comment (andrew-jameson) [inline tdrs-backend/setup.cfg:9]: I omitted `scheduling` to boost coverage, still not sure why it dropped 3% on my two line updates. Codecov says it it was related to files coming from develop that these commits didn't happen. Open to input on how to address this better.

Screenshot 2024-10-11 at 1 49 46 PM

**Review by jtimpe — APPROVED** [review]

**Review by elipe17 — APPROVED** [review]

**Review by ADPennington — APPROVED** [review]
@andrew-jameson feel free to revert the branch changes on this PR. lgtm!

**Pull request merged**